src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -106,15 +154,28 @@ public String toString() {
         return persons.asUnmodifiableObservableList();
     }
 
+    @Override
+    public ObservableList<Company> getCompanyList() {
+        return companies.asUnmodifiableObservableList();
+    }
+
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof AddressBook // instanceof handles nulls
-                && persons.equals(((AddressBook) other).persons));
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof AddressBook)) {
+            return false;
+        }
+
+        AddressBook otherAddressBook = (AddressBook) other;
+        return persons.equals(otherAddressBook.persons)
+                && companies.equals(otherAddressBook.companies);
     }
 
     @Override
     public int hashCode() {
-        return persons.hashCode();
+        return persons.hashCode() + companies.hashCode();
     }
```

Comment:
I think this would be better to do:
return Objects.hash(persons, companies)
because of the nature of hashing functions in Java.